src/database/sqlite/mod.rs

```diff
@@ -729,6 +729,50 @@ impl SqliteConn {
         .expect("Database must be available")
     }
 
+    pub fn list_wallet_transactions(
+        &mut self,
+        txids: &[bitcoin::Txid],
+    ) -> Vec<DbWalletTransaction> {
+        // The UNION will remove duplicates.
+        // We assume that a transaction's block info is the same in every coins row
+        // it appears in.
+        let query = format!(
+            "SELECT t.tx, c.blockheight, c.blocktime \
+            FROM transactions t \
+            INNER JOIN ( \
+                SELECT txid, blockheight, blocktime \
+                FROM coins \
+                WHERE wallet_id = {WALLET_ID} \
+                UNION \
+                SELECT spend_txid, spend_block_height, spend_block_time \
+                FROM coins \
+                WHERE wallet_id = {WALLET_ID} \
+                AND spend_txid IS NOT NULL \
+            ) c ON t.txid = c.txid \
+            WHERE t.txid in ({})",
+            txids
+                .iter()
+                .map(|txid| format!("x'{}'", FrontwardHexTxid(*txid)))
+                .collect::<Vec<_>>()
+                .join(",")
+        );
+        let w_txs: Vec<DbWalletTransaction> =
+            db_query(&mut self.conn, &query, rusqlite::params![], |row| {
+                row.try_into()
+            })
+            .expect("Db must not fail");
+        assert_eq!(
```

Comment:
I added this assertion.

Reply:
Ok. Can we make it a debug_assert? It's not ideal to perform an allocation just for an assertion.

Reply:
Sure, I'll change it.

Reply:
Have changed it now.